# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/SelfHosting.cpp
# Commit: a468786b4570
# Full Hash: a468786b4570997bb28c6ed03acf84f2ffef71df
# Author: Iain Ireland <iireland@mozilla.com>
# Date: 2020-05-22 21:38:53
# Regressor Bug: 1362154
# File Overlap Count: 1
# Description:
#   Bug 1362154: Part 8: Named capture replacements in RegExpGetSubstitution r=arai
#   
#   This patch implements the changes to GetSubstitution to support named captures.
#   
#   See https://tc39.es/proposal-regexp-named-groups/#sec-getsubstitution
# ==============================================================================

diff -r c296b98a4e22 -r a468786b4570 js/src/vm/SelfHosting.cpp
--- a/js/src/vm/SelfHosting.cpp	Thu May 21 04:52:12 2020 +0000
+++ b/js/src/vm/SelfHosting.cpp	Thu May 21 16:57:07 2020 +0000
@@ -1519,7 +1519,7 @@
 static bool intrinsic_RegExpGetSubstitution(JSContext* cx, unsigned argc,
                                             Value* vp) {
   CallArgs args = CallArgsFromVp(argc, vp);
-  MOZ_ASSERT(args.length() == 5);
+  MOZ_ASSERT(args.length() == 6);
 
   RootedArrayObject matchResult(cx, &args[0].toObject().as<ArrayObject>());
 
@@ -1539,9 +1539,12 @@
   int32_t firstDollarIndex = int32_t(args[4].toNumber());
   MOZ_ASSERT(firstDollarIndex >= 0);
 
+  RootedValue namedCaptures(cx, args[5]);
+  MOZ_ASSERT(namedCaptures.isUndefined() || namedCaptures.isObject());
+
   return RegExpGetSubstitution(cx, matchResult, string, size_t(position),
                                replacement, size_t(firstDollarIndex),
-                               args.rval());
+                               namedCaptures, args.rval());
 }
 
 static bool intrinsic_StringReplaceString(JSContext* cx, unsigned argc,
